churaverse-plugins-server/src/churarenPlugin/churarenBossAttackPlugin/churarenBossAttackPlugin.ts

```diff
@@ -0,0 +1,110 @@
+import '@churaverse/churaren-core-plugin-server/event/churarenStartTimerEvent'
+import '@churaverse/player-plugin-server/store/defPlayerPluginStore'
+import '@churaverse/churaren-boss-plugin-server/store/defBossPluginStore'
+import '@churaverse/network-plugin-server/store/defNetworkPluginStore'
+
+export class ChurarenBossAttackPlugin extends BaseGamePlugin {
+  public gameId = CHURAREN_CONSTANTS.GAME_ID
+  private bossAttackPluginStore!: BossAttackPluginStore
+  private mapPluginStore!: MapPluginStore
+
+  public listenEvent(): void {
+    super.listenEvent()
+    this.bus.subscribeEvent('init', this.init.bind(this))
+
+    const socketController = new SocketController(this.bus, this.store)
+    this.bus.subscribeEvent('registerMessage', socketController.registerMessage.bind(socketController))
+    this.bus.subscribeEvent(
+      'registerMessageListener',
+      socketController.setupMessageListenerRegister.bind(socketController)
+    )
+    this.bus.subscribeEvent('registerOnOverlap', this.registerOnOverlap.bind(this))
+  }
+
+  protected subscribeGameEvent(): void {
+    super.subscribeGameEvent()
+    this.bus.subscribeEvent('update', this.update)
+    this.bus.subscribeEvent('bossAttackRequest', this.generateBossAttack)
+    this.bus.subscribeEvent('entitySpawn', this.onSpawnBossAttack.bind(this))
+  }
+
+  protected unsubscribeGameEvent(): void {
+    super.unsubscribeGameEvent()
+    this.bus.unsubscribeEvent('update', this.update)
+    this.bus.unsubscribeEvent('bossAttackRequest', this.generateBossAttack)
+    this.bus.unsubscribeEvent('entitySpawn', this.onSpawnBossAttack.bind(this))
+  }
+
+  protected handleGameStart(): void {}
+
+  protected handleGameTermination(): void {
+    this.bossAttackPluginStore.bossAttacks.clear()
+  }
+
+  private readonly update = (ev: UpdateEvent): void => {
+    moveBossAttacks(ev.dt, this.bossAttackPluginStore.bossAttacks, this.mapPluginStore.mapManager.currentMap)
+    removeDieBossAttack(this.bossAttackPluginStore.bossAttacks, (bossAttackId: string) => {
+      const bossAttackHitMessage = new BossAttackHitMessage({ bossAttackId })
+      const networkPluginStore = this.store.of('networkPlugin')
+      networkPluginStore.messageSender.send(bossAttackHitMessage)
+    })
+  }
+
+  private init(): void {
+    initBossAttackPluginStore(this.store)
+    this.bossAttackPluginStore = this.store.of('bossAttackPlugin')
+    this.mapPluginStore = this.store.of('mapPlugin')
+  }
+
+  private readonly generateBossAttack = (ev: BossAttackRequestEvent): void => {
+    const bossPluginStore = this.store.of('bossPlugin')
+    const boss = bossPluginStore.bosses.get(ev.bossId)
+    const churarenGameInfo = this.store.of('gamePlugin').games.get(this.gameId)
+    const networkPluginStore = this.store.of('networkPlugin')
+    if (churarenGameInfo === undefined || boss === undefined) return
+
+    // ボスの攻撃をフロントに送信
+    sendSpawnedBossAttack(networkPluginStore.messageSender, this.bus, boss.position, boss.bossId)
```

Comment:
こちらもまとめられそうです。
```suggestion
    if (churarenGameInfo === undefined || boss === undefined) return

    // ボスの攻撃をフロントに送信
    sendSpawnedBossAttack(this.store.of('networkPlugin').messageSender, this.bus, boss.position, boss.bossId)
```

また、仮に変数で置く場合でも基本的に変数は使う直前に宣言した方がスコープが小さくなって見やすいです
```suggestion
    if (churarenGameInfo === undefined || boss === undefined) return

    // ボスの攻撃をフロントに送信
    const networkPluginStore = this.store.of('networkPlugin')
    sendSpawnedBossAttack(networkPluginStore.messageSender, this.bus, boss.position, boss.bossId)
```